File: python/424_替换后的最长重复字符_medium.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        n = len(s)
        left, right = 0, 0
        idx = collections.defaultdict(int)
        ans = 0
        countMax = 0
        while right < n:
            idx[ord(s[right])-ord('A')] += 1
            countMax = max(countMax, idx[ord(s[right])-ord('A')])
            right += 1        
            if right-left > countMax + k:
                idx[ord(s[left])-ord('A')] -= 1
                left += 1
            ans = max(ans, right-left)
        return ans

File: cpp/424_替换后的最长重复字符_medium.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n = s.length();
        int maxCount = 0;
        int left = 0;
        int right = 0;
        vector<int> num(26, 0);
        int ans = 0;
        while (right < n) {
            num[s[right]-'A']++;
            maxCount = max(maxCount, num[s[right]-'A']);
            right++;
            if (right-left > maxCount+k) {
                num[s[left]-'A']--;
                left++;
            }
            ans = max(ans, right-left);
        }
        return ans;
    }
};